File: backend/src/main/java/com/quizmatrix/quizmatrix/model/LeaderBoard.java
package com.quizmatrix.quizmatrix.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "leaderboard_domain")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class LeaderBoard {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Integer id_leaderboard;
    private Integer id_user;
    private Integer id_domain;
    private Integer id_quiz;
    private Integer score;
    private Integer time;

}

File: frontend/QuizMatrix/src/app/models/leaderboard.ts
export class Leaderboard {
  id_leaderboard: number;
  id_user: number;
  id_domain: number;
  id_quiz: number;
  score: number;
  time: number;

  constructor(
    id_user: number,
    id_domain: number,
    id_quiz: number,
    score: number,
    time: number,
    id_leaderboard: number = 0
  ) {
    this.id_leaderboard = id_leaderboard;
    this.id_user = id_user;
    this.id_domain = id_domain;
    this.id_quiz = id_quiz;
    this.score = score;
    this.time = time;
  }
}
